www/guides/react.md

```diff
@@ -0,0 +1,368 @@
+
+Now that you have a basic project up and running, to install React, run the following command in your project directory:
+
+```bash
+npm install react react-dom --save
+```
+
+> 💡 Tip: in this guide commands use `--save` for frontend libraries and `--save-dev` for your developer tools to keep them organized separately. This is totally up to you, though.
+
+## Create your first React component
+
+React relies on a special templating language called JSX. Snowpack has built in support for JSX in files with the `.jsx` extension. That means that there's no plugins or configuration needed to write your first React component. Rename `index.js` file to `index.jsx` so that Snowpack knows it's now a JSX file:
+
+```bash
+mv index.js index.jsx
+```
+
+> 💡 Tip: you do not need to update your `index.html` script tag to point to `index.jsx`. Because JSX isn't supported in the browser, Snowpack builds your file to a runnable `.js` file. This is good to remember when you're referencing built files in HTML `<script src="">` and `<link href="">` elements.
+
+You can now import React in `index.jsx` and add a simple test component just to make sure it's working:
+
+```diff
+/* Add JavaScript code here! */
+- console.log('Hello World! You did! Welcome to Snowpack :D');
++import React from 'react';
++import ReactDOM from 'react-dom';
++ReactDOM.render(<div>"HELLO REACT"</div>, document.getElementById('root'));
+```
+
+Since the React code is rendering into an element with the ID `root`, you'll need to add that to `index.html`:
+
+```diff
+  <body>
+-    <h1>Welcome to Snowpack!</h1>
++    <div id="root"></div>
+    <script type="module" src="/index.js"></script>
+```
+
+<div class="frame"><img src="/img/guides/react/minimalist-hello-world-react.png" alt="screenshot of the project, which shows 'HELLO REACT' on a white background" class="screenshot"/></div>
+
+You've just created your first React component in Snowpack!
+
+## Customize your project layout
+
+Since you'll be adding a bunch of new files, you probably don't want them crowding up your top-level root directly. Snowpack is flexible enough to support whatever project layout that you prefer. In this guide, you'll learn how to use a popular project pattern from the React community.
+
+```
+📁 src : your React components and their assets (CSS, images)
+↳ index.jsx
+📁 public : global assets like images, fonts, icons, and global CSS
+↳ index.css
+↳ index.html
+```
+
+Use your favorite visual editor to rearrange and rename, or run these commands in the terminal:
+
+```bash
+mkdir src
+mkdir public
+mv index.jsx src/index.jsx
+mv index.html public/index.html
+mv index.css public/index.css
+```
+
+This changes how Snowpack builds the files since by default Snowpack scans all directories:
+
+```
+/ (builds to → /)
+📁 src
+↳ index.jsx (builds to → src/index.js)
+📁 public
+↳ index.css (builds to → public/index.css)
+↳ index.html (builds to → public/index.html)
+```
```

Comment:
```suggestion
├── 📁 src/
│   └── index.jsx → src/index.js
└── 📁 public/
    ├── index.css → public/index.css
    └── index.html → public/index.html
```
Reading this again in the guide with everything else going on, it was hard for me to see this as a tree structure. I think using the “box drawing” unicode characters, and the convention of two dashes per file (`──`) make the file structure quicker to recognize at-a-glance

Reply:
Also, I think you could just use `→` in the tree structure, and add a note below like “→ represents the file that Snowpack will build (note in some cases Snowpack will overwrite what’s there)”